addons/token-exchange/blue_secret_controller.go

```diff
@@ -0,0 +1,190 @@
+package addons
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/red-hat-storage/odf-multicluster-orchestrator/controllers/secretutils"
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1lister "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+const (
+	RookType                     = "kubernetes.io/rook"
+	DefaultBlueSecretMatchString = "cluster-peer-token"
+)
+
+var (
+	blueSecretMatchString string
+)
+
+type blueSecretTokenExchangeAgentController struct {
+	hubKubeClient     kubernetes.Interface
+	hubSecretLister   corev1lister.SecretLister
+	spokeKubeClient   kubernetes.Interface
+	spokeSecretLister corev1lister.SecretLister
+	clusterName       string
+	recorder          events.Recorder
+	kubeconfig        *rest.Config
+}
+
+func newblueSecretTokenExchangeAgentController(
+	hubKubeClient kubernetes.Interface,
+	hubSecretInformers corev1informers.SecretInformer,
+	spokeKubeClient kubernetes.Interface,
+	spokeSecretInformers corev1informers.SecretInformer,
+	clusterName string,
+	recorder events.Recorder,
+	kconfig *rest.Config,
+) factory.Controller {
+	c := &blueSecretTokenExchangeAgentController{
+		hubKubeClient:     hubKubeClient,
+		hubSecretLister:   hubSecretInformers.Lister(),
+		spokeKubeClient:   spokeKubeClient,
+		spokeSecretLister: spokeSecretInformers.Lister(),
+		clusterName:       clusterName,
+		recorder:          recorder,
+		kubeconfig:        kconfig,
+	}
+	klog.Infof("Creating blue secret controller")
+
+	blueSecretMatchString = os.Getenv("TOKEN_EXCHANGE_BLUE_SECRET_STRING")
+	if "" == blueSecretMatchString {
```

Comment:
Mostly seen it other way round, that is , ` if blueSecretMatchString == "" `

Reply:
This form is https://en.wikipedia.org/wiki/Yoda_conditions . I changed it the other way to maintain uniformity.